src/pubsub/mod.rs

```diff
@@ -77,14 +81,14 @@ impl MessageValidator {
         }
 
         // calculate and return the end to end latency
-        let ts = u64::from_be_bytes([v[16], v[17], v[18], v[19], v[20], v[21], v[22], v[23]]);
-        let latency = now_unix - UnixInstant::from_nanos(ts);
+        let ts = u64::from_be_bytes(v[16..24].try_into().unwrap());
+        let latency = now_unix.duration_since(UNIX_EPOCH).unwrap().as_nanos() as u64 - ts;
```

Comment:
Does the fact that ts is in big-endian form and now_unit presumably in little-endian mess with this subtraction?